core/common/src/main/java/alluxio/collections/IndexedSet.java

```diff
@@ -128,21 +139,82 @@
   }
 
   /**
+   * An interface extending {@link FieldIndex}, represents a unique index. A unique index is an
+   * index where each index value only maps to one object.
+   *
+   * @param <T> type of objects in this {@link IndexedSet}
+   */
+  public interface UniqueFieldIndex<T> extends FieldIndex<T> {
```

Comment:
@jsimsa Yes, it should. But as I replied to your previous comment, if we make the FieldIndex private, those who call the function 
`public IndexedSet(FieldIndex<T> field, FieldIndex<T>... otherFields)`
will not be able to access the argument type. I haven't  figured out a solution so finally I left them public.